clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -171,6 +195,7 @@ public void removeSensor(String name) {
                     }
                 }
             }
```

Comment:
Isn't that implicit though? If a parent sensor reports "hasExpired == true", then it has not received a record and neither have any of its children